src/main/java/org/csanchez/jenkins/plugins/kubernetes/StandardPlannedNodeBuilder.java

```diff
@@ -1,16 +1,29 @@
 package org.csanchez.jenkins.plugins.kubernetes;
 
-import hudson.model.Computer;
+import com.google.common.util.concurrent.Futures;
+import hudson.model.Descriptor;
 import hudson.slaves.NodeProvisioner;
 
+import java.io.IOException;
+
 /**
  * The default {@link PlannedNodeBuilder} implementation, in case there is other registered.
  */
 public class StandardPlannedNodeBuilder extends PlannedNodeBuilder {
     @Override
     public NodeProvisioner.PlannedNode build() {
-        return new NodeProvisioner.PlannedNode(getTemplate().getDisplayName(),
-                Computer.threadPoolForRemoting.submit(new ProvisioningCallback(getCloud(), getTemplate())),
-                getNumExecutors());
+        KubernetesCloud cloud = getCloud();
+        PodTemplate t = getTemplate();
+        try {
+            return new NodeProvisioner.PlannedNode(t.getDisplayName(),
+                    Futures.immediateFuture(KubernetesSlave
+                            .builder()
+                            .podTemplate(cloud.getUnwrappedTemplate(t))
+                            .cloud(cloud)
+                            .build()),
+                    getNumExecutors());
+        } catch (IOException | Descriptor.FormException e) {
+            throw new RuntimeException(e);
```

Comment:
To retain original behavior, this should actually return a failing `Future`. But are these checked exceptions actually possible to begin with?